app/decorators/submission_trait_scores_upload_decorator.rb

```diff
@@ -12,14 +12,47 @@ def parser_summary
           group_by{ |id, scores| scores.size }.
           sort{ |score1, score2| score2.first <=> score1.first }
 
-        summary << " - parsed #{trait_scores.size} plant(s) with unique identification"
+        summary << " - parsed #{trait_scores.size} plant scoring unit(s) with unique identification"
         histogram.each do |scoring_number, plants|
-          summary << "  - #{plants.size} plant(s) have #{scoring_number} trait score(s) recorded"
+          summary << "  - #{plants.size} unit(s) have #{scoring_number} trait score(s) recorded"
         end
       end
 
+      accessions = object.submission.content.step04.accessions
```

Comment:
Ok, that refactoring will do.

I did some research regarding rendering partials to string, so I will share anyway. It seems it's not that easy to do that in a decorator unless you pass controller as the [context](https://github.com/drapergem/draper#adding-context). Otherwise, you can use `ActionView::PartialRenderer` directly but I am not sure if it is considerer public API, probably not (and it needs somewhat magical arguments).

```ruby
class Decorator < Draper::Decorator
  def render_errors
    ActionView::PartialRenderer.new(h.lookup_context).render(h, { partial: "/errors", locals: { errors: { :foo => "foo!" } } }, nil)
  end
end
```


Another option would be to use erb or haml directly.

```ruby
engine = Haml::Engine.new("%p Haml code!")
engine.render #=> "<p>Haml code!</p>\n"
```